clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/MetricsDuringTopicCreationDeletionTest.java

```diff
@@ -0,0 +1,194 @@
+import org.apache.kafka.server.metrics.KafkaYammerMetrics;
+
+import com.yammer.metrics.core.Gauge;
+
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MetricsDuringTopicCreationDeletionTest {
+
+    private static final String TOPIC_NAME_PREFIX = "topic";
+    private static final int TOPIC_NUM = 2;
+    private static final int CREATE_DELETE_ITERATIONS = 3;
+    private static final int REPLICATION_FACTOR = 1;
+    private static final int PARTITION_NUM = 3;
+
+    private final ClusterInstance clusterInstance;
+    private final List<String> topics;
+    private volatile boolean running = true;
+
+    private int initialOfflinePartitionsCount = 0;
+    private int initialPreferredReplicaImbalanceCount = 0;
+    private int initialUnderReplicatedPartitionCount = 0;
+
+    public MetricsDuringTopicCreationDeletionTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+        this.topics = new ArrayList<>();
+        for (int n = 0; n < TOPIC_NUM; n++) {
+            topics.add(TOPIC_NAME_PREFIX + n);
+        }
+    }
+
+    /*
+     * Captures initial values of key controller metrics.
+     * These will be compared with final values to detect any changes.
+     */
+    @BeforeEach
+    public void setUp() {
+        try {
+            initialOfflinePartitionsCount = getGauge("OfflinePartitionsCount").value();
+            initialPreferredReplicaImbalanceCount = getGauge("PreferredReplicaImbalanceCount").value();
+            initialUnderReplicatedPartitionCount = getGauge("UnderReplicatedPartitions").value();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Closeable runThread() {
+        var closed = new AtomicBoolean(false);
+        var f = CompletableFuture.runAsync(() -> {
+            while (!closed.get()) {
+                clusterInstance.brokers().values().forEach(broker -> {
+                    if (running) {
+                        // Get UnderReplicatedPartitions through JMX
+                        Optional<Integer> underReplicatedCount = KafkaYammerMetrics.defaultRegistry().allMetrics().entrySet().stream()
+                                .filter(entry -> entry.getKey().getName().endsWith("UnderReplicatedPartitions"))
+                                .map(entry -> ((Gauge<Integer>) entry.getValue()).value())
+                                .findFirst();
+
+                        int count = underReplicatedCount.orElse(0);
+                        if (count != initialUnderReplicatedPartitionCount) {
+                            running = false;
+                        }
+                    }
+                });
```

Comment:
This block doesn't really use `broker` variable. We can simplify it as following:

```suggestion
                if (running) {
                    // Get UnderReplicatedPartitions through JMX
                    Optional<Integer> underReplicatedCount = KafkaYammerMetrics.defaultRegistry().allMetrics().entrySet().stream()
                        .filter(entry -> entry.getKey().getName().endsWith("UnderReplicatedPartitions"))
                        .map(entry -> ((Gauge<Integer>) entry.getValue()).value())
                        .findFirst();

                    int count = underReplicatedCount.orElse(0);
                    if (count != initialUnderReplicatedPartitionCount) {
                        running = false;
                    }
                }
```